### ISSUE TITLE
Evaluate and offer an alternative ontology browser

### ISSUE THREAD

**Issue by lewismc** [ESIPFed/cor#67]
**Q**: Did you also look at the WebProtege tool? 
 **A**: (John G: Not sure for what purpose this question was meant as COR is not intended as an ontology editor. 

**Q**: It's worthwhile reviewing TopBraid Composer, MagicDraw, Neo4j, Hozo editor, and other similar semantic tools. Their free versions are powerful, and some are open source. Protégé is not as powerful, and doesn't have functionalities these other tools do. 
**A**: Definitely worth evaluating other tools. John G: Having seen Protégé and WebProtégé up close, I don’t think it’s fair to say they are “not as powerful”—they are differently powerful and offer an extensive set of unique features and plugins.

**Comment by lewismc** [comment]
The following comments are extracted from another ESIP Summer 2020 session titled [Can we better utilise and extend the ESIP Community Ontology Repository (COR)? ](https://docs.google.com/document/d/1TzQExAe4Zb7fcHsyXRjOdB2R1ib8pVJ9lMhWbrbLgl4/edit)

**Comment by lewismc** [comment]
Seeing as we use AllegroGraph, we could likely easily provision Gruff
https://allegrograph.com/products/gruff/

**Comment by rrovetto** [comment]
Per the call today, attached is a document listing some tools
[Ontology Graph Editor Tools_v1_Rovetto.docx](https://github.com/ESIPFed/cor/files/5484564/Ontology.Graph.Editor.Tools_v1_Rovetto.docx)
And [this other document](https://esip-all.slack.com/files/USVQYKYNM/F01823F3648/info_kos-ontorepo_rovetto_v1.docx) in a July ESIP post.

**Comment by lewismc** [comment]
Thank you @rrovetto 👍 

@brandonnodnarb please let me know when you want to have a session to begin attacking this one.

**Comment by brandonnodnarb** [comment]
Technically on leave until the 10th but a few quick thoughts:
1) Do we need to separate pure viewers from tools/stacks which have increased functionality?  It seems like the unknowns are many (w.r.t. community needs and wants) and the functionality is inconsistent which makes it difficult to compare like for like.
2) Are there specific "views" that need prioritizing --- e.g. graph vs tree?
3) I looked briefly at WebProtege yesterday (again); it's not necessarily for viz, but it is an alternative view to that of COR.  Might be a reasonable place to start with the added benefit of passively starting a collaborative management tool discussion.

**Comment by graybeal** [comment]
example visualization from WebProtege (an old version of SWEET, not updated for a while, just something I had lying around).

![Screen Shot 2020-11-08 at 8 27 40 PM](https://user-images.githubusercontent.com/1266896/98502424-d1cdb680-2206-11eb-98df-4e66d25c01d6.png)

I think you can see that with this link, at least if you have a web protege account:
  https://webprotege.stanford.edu/#projects/aec8c830-2821-4c56-bcf3-6d2c0ee5581b/edit/Classes?selection=Class(%3Chttp://sweetontology.net/phenGeol/Metamorphic%3E)

**Comment by brandonnodnarb** [comment]
A discussion about viewing SWEET using WebVOWL on the [sweetontology slack channel](https://esip-all.slack.com/archives/CLQ6DJ669/p1610147197008700) is relevant to this issue.

Aside from the loading issue, a customized, local version of WebVOWL may be a "quick" way to get user feedback.  Perhaps showing something like this at the WM is in order?

**Comment by lewismc** [comment]
@brandonnodnarb and I brainstormed how we are going to move forward with this ticket. We both have an understanding of how Gruff (for the time being) can be configured to connect to COR's SPARQL endpoint. 
1. augment COR docker-compose with Gruff service. This would deploy Gruff to http://cor.esipfed.org/gruff
2. provide a short YouTube video demonstrating how to use Gruff to connect to COR and browse data
3. perform an evaluation period for Gruff and get community feedback
4. use feedback to (i) send to Franz to inform them of community usage at ESIP, and (ii) determine whether we want to support Gruff moving forward
5. Once this is done... we 'may' move onto evaluate some other tooling.

**Comment by carueda** [comment]
@lewismc @brandonnodnarb 

Nice goal.  Incorporating gruff has been a desire for a long time. I don't remember the specific details but we (I) ended up settling on using their Webview interface (http://cor.esipfed.org/webview/ --included with Agraph itself-- which, in any case, is more for advanced use).

BTW, also to note that COR/ORR still uses a pretty old AGraph version.  Some attempts (a year+ ago?) to explore more recent versions and level of effort to update dependent code on the ORR were not very fruitful. We put in place a testing/staging space to explore this but had to shut it down at some point, I think because of lack of enough space and/or cpu on the machine.  I think it's worth considering a strategy like this again.

**Comment by brandonnodnarb** [comment]
GCMD has a new[ 'beta' viewer](https://gcmd.earthdata.nasa.gov/KeywordViewer/scheme/all/).  Might be worth having a look. It's clean...

**Comment by graybeal** [comment]
Is it a triples viewer, or just a GCMD viewer? Their native format did not used to follow a standard (long ago).

**Comment by lewismc** [comment]
@graybeal 

> ... or just a GCMD viewer?

Yes

**Comment by brandonnodnarb** [comment]
Agreed.  I had hopes it was something else, but it's a *custom* viewer.

**Comment by lewismc** [comment]
We need to investigate whether Gruff is compatible with [AllegroGraph 6.X which is what is run in in the COR deployment](https://github.com/mmisw/orr-instance-template/blob/master/docker-compose.yml#L30-L40).

**Comment by carueda** [comment]
As just mentioned in the today's meeting, if just connecting to a SPARQL endpoint from Gruff in a generic way, then it wouldn't matter that much what's running behind COR's for the endpoint, unless there's some special connection that would allow Gruff to exploit more Allegrograph specific mechanisms (or for performance reasons, etc.)

**Comment by lewismc** [comment]
Cleaning up stale issues 😄
